panther_manager/test/plugins/include/plugin_test_utils.hpp

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PANTHER_MANAGER_PLUGIN_TEST_UTILS_HPP_
+#define PANTHER_MANAGER_PLUGIN_TEST_UTILS_HPP_
+
+#include <functional>
+#include <limits>
+#include <map>
+#include <memory>
+#include <string>
+#include <thread>
+
+#include <behaviortree_cpp/bt_factory.h>
+#include <gtest/gtest.h>
+#include <rclcpp/rclcpp.hpp>
+
+#include <std_srvs/srv/set_bool.hpp>
+#include <std_srvs/srv/trigger.hpp>
+
+#include <panther_msgs/srv/set_led_animation.hpp>
+
+#include <panther_manager/plugins/action/call_set_bool_service_node.hpp>
+#include <panther_manager/plugins/action/call_set_led_animation_service_node.hpp>
+#include <panther_manager/plugins/action/call_trigger_service_node.hpp>
+#include <panther_manager/plugins/action/shutdown_hosts_from_file_node.hpp>
+#include <panther_manager/plugins/action/shutdown_single_host_node.hpp>
+#include <panther_manager/plugins/action/signal_shutdown_node.hpp>
+#include <panther_manager/plugins/decorator/tick_after_timeout_node.hpp>
+
+namespace panther_manager::plugin_test_utils
+{
+
+struct BehaviorTreePluginDescription
+{
+  std::string name;
+  std::map<std::string, std::string> params;
+};
+
+class PluginTestUtils : public testing::Test
+{
+public:
+  virtual void SetUp() override final;
+
+  virtual void TearDown() override final;
+
+  std::string BuildBehaviorTree(
+    const std::string & plugin_name, const std::map<std::string, std::string> & service,
+    double tick_after_timeout);
+
+  void CreateTree(
+    const std::string & plugin_name, const std::map<std::string, std::string> & service,
+    double tick_after_timeout = std::numeric_limits<double>::quiet_NaN());
+
+  BT::Tree & GetTree();
+
+  BT::BehaviorTreeFactory & GetFactory();
+
+  template <typename Service, typename Request, typename Response>
+  void CreateService(
+    const std::string & service_name,
+    std::function<void(std::shared_ptr<Request>, std::shared_ptr<Response>)> service_callback)
```

Comment:
```suggestion
  template <typename ServiceT>
  void CreateService(
    const std::string & service_name,
    std::function<void(const ServiceT::Request::SharedPtr, ServiceT::Response::SharedPtr)> service_callback)
```
this simplifies method calling

Reply:
Thank you. It worked!